app/frontend/controllers/locale_controller.js

```diff
@@ -0,0 +1,133 @@
+  window.i18nHelper = new I18nHelper();
+
+let LanguageDetector = null;
+
+  if (platformHelper.isElectron()) {
+    LanguageDetector = require('i18next-electron-language-detector');
+  } else {
+    const _LanguageDetector = require('../platform/i18next_browser_language_detector.js');
+    LanguageDetector = new _LanguageDetector();
+  }
+
+  await i18n
+    .use(LanguageDetector)
+    .use(I18nIpcBackend)
+    .init(i18nextOptions);
+
+  systemLocale = i18n.language;  
+
+  jqueryI18next.init(i18n, $, {
+    tName: 't', // --> appends $.t = i18next.t
+    i18nName: 'i18n', // --> appends $.i18n = i18next
+    handleName: 'localize', // --> appends $(selector).localize(opts);
+    selectorAttr: 'i18n', // selector for translating elements
+    targetAttr: 'i18n-target', // data-() attribute to grab target element to translate (if different than itself)
+    optionsAttr: 'i18n-options', // data-() attribute that contains options, will load/set if useOptionsAttr = true
+    useOptionsAttr: false, // see optionsAttr
+    parseDefaultValueFromContent: true // parses default values from content ele.val or ele.text
+  });
+
+  if (await ipcSettings.has(SETTINGS_KEY)) {
+    await i18n.changeLanguage(await ipcSettings.get(SETTINGS_KEY, FALLBACK_LOCALE));
+  }
+
+  if (platformHelper.isTest()) { // Use English for test mode
+    await i18n.changeLanguage('en');
+  }
+
+  window.reference_separator = i18n.t('general.chapter-verse-separator');
+}
+
+async function changeLocale(newLocale) {
+
+  await i18n.changeLanguage(newLocale);
+
+  await ipcSettings.set(SETTINGS_KEY, newLocale);
+
+  $(document).localize();
+
+  // Todo: Bind to event in respective dependent components instead
+  window.reference_separator = i18n.t('general.chapter-verse-separator');
+  await app_controller.book_selection_menu.localizeBookSelectionMenu();
+  await app_controller.assign_last_tag_button.updateLabel();
+  await app_controller.verse_selection.updateSelectedVersesLabel();
+  await app_controller.tab_controller.updateTabTitleAfterLocaleChange();
+  app_controller.tab_controller.localizeTemplate();
+  await app_controller.updateTagsView(undefined, true);
+  tags_controller.refreshTagDialogs();
+
+  await notifySubscribers(newLocale);
+}
+
+var localeSubscribers = [];
+function onChangeLocale(subscriberCallback) {
+  if (typeof subscriberCallback === 'function') {
+    localeSubscribers.push(subscriberCallback);
+  }
+}
+
+async function notifySubscribers(locale) {
+  for (let subscriberCallback of localeSubscribers) {
+    await subscriberCallback(locale);
+  }
+}
+
+async function detectLocale() {
+  await changeLocale(systemLocale || FALLBACK_LOCALE);
+}
+
+function getAvailableLocales() {
+  return AVAILABLE_LOCALES.sort();
+}
+
+
+
+module.exports.initI18N = initI18N;
+module.exports.changeLocale = changeLocale;
+module.exports.detectLocale = detectLocale;
+module.exports.onChangeLocale = onChangeLocale;
+module.exports.getAvailableLocales = getAvailableLocales;
```

Comment:
I read somewhere that it's better to not overwrite default `module.exports` object.
```javascript
module.exports = {
  initI18N,
  changeLocale,
  detectLocale,
  onChangeLocale,
  getAvailableLocales
};
```
is **not** the best practice